src/modules/chat/actions/types.jsx

```diff
@@ -354,6 +354,40 @@ export const untimeout = {
 	}
 }
 
+export const highlight = {
+	presets: [{
+		appearance: {
+			type: 'icon',
+			icon: 'ffz-i-eye'
+		},
+	}],
+
+	defaults: {},
+
+	required_context: ['user'],
+
+	title: 'Highlight User',
+
+	tooltip(data) {
+		return this.i18n.t('chat.actions.highlight.tooltip', `Highlight ${data.user.login}`);
+	},
+
+	click(event, data) {
+		let key = 'chat.filtering.highlight-temp'
+		let val = this.settings.provider.get(key)
+		if(!val)
+			val = []
+		if(val.includes(data.user.id))
+			val = val.filter(value => value !== data.user.id);
+		else
+			val.push(data.user.id)
+		this.settings.provider.set(key, val)
+		this.resolve('chat').emit('chat:update-lines')
```

Comment:
This bit is fine, but this seemed like a good place to point out that settings are synced in real time between browser tabs with FrankerFaceZ. You might want to set up code that triggers on other tabs when the provider updates your list so that those tabs will update their chat lines.

See [overrides](https://github.com/FrankerFaceZ/FrankerFaceZ/blob/master/src/modules/chat/overrides.js#L39) or [addons](https://github.com/FrankerFaceZ/FrankerFaceZ/blob/master/src/addons.js#L76) for an example on how that's done.